### PR TITLE
Feature/weee sent on edit data

### PR DESCRIPTION
**Pull request by SamGoodeySFW**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/EditSentOnAatfSiteHandler.cs [hunk 1]
```diff
@@ -16,40 +21,68 @@ public class EditSentOnAatfSiteHandler : IRequestHandler<EditSentOnAatfSite, Gui
...
             this.organisationDetailsDataAccess = orgDataAccess;
+            this.offSiteDataAccess = offSiteDataAccess;
         }
 
         public async Task<Guid> HandleAsync(EditSentOnAatfSite message)
         {
             authorization.EnsureCanAccessExternalArea();
 
-            var sentOn = await genericDataAccess.GetById<WeeeSentOn>(message.WeeeSentOnId);
+            if (message.SiteAddressData != null)
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/EditSentOnAatfSiteHandler.cs]: Probably want a test for this null check

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetSentOnOperatorSiteHandler.cs [hunk 2]
```diff
@@ -28,7 +28,12 @@ public async Task<AatfAddressData> HandleAsync(GetSentOnOperatorSite message)
 
             var aatfAddress = await getSentOnAatfSiteDataAccess.GetWeeeSentOnOperatorAddress(message.WeeeSentOnId);
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetSentOnOperatorSiteHandler.cs]: This class has no covering tests as far as I can see?

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetWeeeSentOnHandler.cs [hunk 3]
```diff
@@ -42,14 +42,29 @@ public async Task<List<WeeeSentOnData>> HandleAsync(GetWeeeSentOn message)
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetWeeeSentOnHandler.cs]: Apparently the tests for this class don't cover most of the between 38 and 65.  Perhaps in the tests you need to have items in the sentOnList in HandleAsync_GivenGetSentOnAatfSiteRequest_AddressDataShouldBeMapped()

File: src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnAndAatfToSentOnCreateSiteViewModelMap.cs [hunk 4]
```diff
@@ -22,12 +23,20 @@ public SentOnCreateSiteViewModel Map(ReturnAndAatfToSentOnCreateSiteViewModelMap
         {
             Guard.ArgumentNotNull(() => source, source);
 
+            if (source.SiteAddressData == null)
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnAndAatfToSentOnCreateSiteViewModelMap.cs]: Probably need a test on this null check; the contents of the if statement are not covered

File: src/EA.Weee.Web/Areas/AatfReturn/Requests/AddSentOnAatfSiteRequestCreator.cs [hunk 5]
```diff
@@ -1,12 +1,24 @@
...
+    using System;
 
     public class AddSentOnAatfSiteRequestCreator : IAddSentOnAatfSiteRequestCreator
     {
-        public AddSentOnAatfSite ViewModelToRequest(SentOnCreateSiteViewModel viewModel)
+        public SentOnAatfSite ViewModelToRequest(SentOnCreateSiteViewModel viewModel)
         {
+            if (viewModel.Edit)
+            {
+                return new EditSentOnAatfSite()
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Requests/AddSentOnAatfSiteRequestCreator.cs]: This doesn't have a covering test for the Edit = true case

File: src/EA.Weee.Web/Areas/AatfReturn/Requests/EditSentOnAatfSiteRequestCreator.cs [hunk 6]
```diff
@@ -2,12 +2,23 @@
...
 
     public class EditSentOnAatfSiteRequestCreator : IEditSentOnAatfSiteRequestCreator
     {
-        public EditSentOnAatfSite ViewModelToRequest(SentOnCreateSiteOperatorViewModel viewModel)
+        public SentOnAatfSite ViewModelToRequest(SentOnCreateSiteOperatorViewModel viewModel)
         {
-            var aatfSite = new EditSentOnAatfSite()
+            if (viewModel.OperatorAddressFound)
+            {
+                return new EditSentOnAatfSite()
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Requests/EditSentOnAatfSiteRequestCreator.cs]: No covering tests for OperatorAddressFound = true

File: src/EA.Weee.Web/Areas/AatfReturn/Requests/ObligatedSentOnWeeeRequestCreator.cs [hunk 7]
```diff
@@ -25,7 +25,15 @@ public override ObligatedBaseRequest ViewModelToRequest(ObligatedViewModel viewM
                         householdValue,
                         nonHouseholdValue));
             }
-            
+
+            if (viewModel.Edit)
+            {
+                return new EditObligatedSentOn()
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Requests/ObligatedSentOnWeeeRequestCreator.cs]: No test for edit=true

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetSentOnOperatorSiteHandler.cs (hunk 2 above)
Comment (SamGoodeySFW) [reply src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetSentOnOperatorSiteHandler.cs]: I think I had previously used this request but is no longer needed, will delete.

File: src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetWeeeSentOnHandler.cs [hunk 8]
```diff
@@ -42,14 +42,29 @@ public async Task<List<WeeeSentOnData>> HandleAsync(GetWeeeSentOn message)
...
+                if (item.OperatorAddress != null)
+                {
+                    weeeSentOnData.OperatorAddress = addressMapper.Map(item.OperatorAddress);
+                    weeeSentOnData.OperatorAddressId = item.OperatorAddress.Id;
+                }
+
                 weeeSentOnList.Add(weeeSentOnData);
             }
 
+            if (message.WeeeSentOnId != null)
```
Comment (JB12298) [inline src/EA.Weee.RequestHandlers/AatfReturn/ObligatedSentOn/GetWeeeSentOnHandler.cs]: Need a test here for the null check
